src/ch/ehi/ili2db/fromili/TypeUtility.java

```diff
@@ -0,0 +1,172 @@
+                " SET "+DbNames.META_INFO_TABLE_TAB_SETTING_COL+" = ? WHERE "
+                +DbNames.META_INFO_TABLE_TAB_TABLENAME_COL+" = ? AND "
+                +DbNames.META_INFO_TABLE_TAB_TAG_COL+" = '"+DbExtMetaInfo.TAG_COL_TYPES+"';";
+
+            PreparedStatement prepStmt = conn.prepareStatement(stmt);
+            prepStmt.setString(1, classesString);
+            prepStmt.setString(2, table);
+            prepStmt.execute();
+        }catch(SQLException ex){
+			throw new Ili2dbException("Failed to add type column property",ex);
+        }
+    }
+
+    public static void addTypeConstraints(Connection conn, String dbSchema, NameMapping mapping)
+        throws Ili2dbException {
+        /**
+         * Adds a check constraint on the t_type column.
+         */
+
+        ArrayList<String> tables = getTablesWithTypeSetting(conn, dbSchema);
+        for (String table : tables) {
+            addTypeConstraint(conn, dbSchema, table, getSubClasses(conn, dbSchema, mapping.mapSqlTableName(table)));
+        }
+    }
+
+    private static void addTypeConstraint(Connection conn, String dbSchema,  String table, ArrayList<String> subClasses)
+        throws Ili2dbException {
+
+        String classesString = "";
+        for (String s: subClasses){
+            classesString += "'"+s+"', ";
+        }
+        //Remove last comma and space
+        classesString = classesString.substring(0, classesString.length() - 2);
+
+        try{
+            String stmt = "ALTER TABLE "+dbSchema+"."+table+
+                " ADD CONSTRAINT type_constraint CHECK ("+DbNames.T_TYPE_COL+
+                " IN ("+classesString+"));";
+
+            PreparedStatement prepStmt = conn.prepareStatement(stmt);
+            prepStmt.execute();
+        }catch(SQLException ex){
+			throw new Ili2dbException("Failed to add type constraint ",ex);
+        }
+    }
+
+    private static ArrayList<String> getSubClasses(Connection conn, String dbSchema, String baseClass)
+        throws Ili2dbException {
+
+        ArrayList<String> result = new ArrayList<String>();
+        result.add(getSqlNameFromIliname(conn, dbSchema, baseClass));
+        try{
+			String stmt= "SELECT "+DbNames.INHERIT_TAB_THIS_COL+" FROM "
+                +dbSchema+"."+DbNames.INHERIT_TAB+" WHERE "
+                +DbNames.INHERIT_TAB_BASE_COL+"= ?";
+
+			PreparedStatement prepStmt = conn.prepareStatement(stmt);
+            prepStmt.setString(1, baseClass);
+            ResultSet rs = prepStmt.executeQuery();
+			try{
+				while(rs.next()){
+					result.add(getSqlNameFromIliname(conn, dbSchema, rs.getString(1)));
+				}
+			}catch(SQLException ex){
+				throw ex;
+			}finally{
+			    rs.close();
+				prepStmt.close();
+			}
+		}catch(SQLException ex){
+			throw new Ili2dbException("Failed to read inheritance table ",ex);
+		}
+    return result;
+    }
+
+    private static ArrayList<String> getTablesWithTypeSetting(Connection conn, String dbSchema)
```

Comment:
If you create the constraint and metainfo where the t_type column is created, you don't need to evaluate if a t_ype column exists...